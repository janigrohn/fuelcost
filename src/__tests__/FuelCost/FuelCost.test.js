import React from 'react';
import { render, fireEvent, cleanup, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FuelCost from '../../Components/FuelCost/FuelCost';

afterEach(cleanup);

const testTogglingFor = async (name) => {
	const { getByTestId, queryByTestId, queryAllByTestId } = render(<FuelCost />);
	const overlay = getByTestId('overlay');
	// Check initial state
	expect(overlay).toHaveStyle('pointer-events: none');
	expect(queryByTestId(`${name}-full`)).not.toBeInTheDocument();
	// Expand
	fireEvent.click(getByTestId(`${name}-expand`));
	expect(overlay).toHaveStyle('pointer-events: auto');
	expect(queryAllByTestId(`${name}-full`)).toHaveLength(2);
	expect(queryAllByTestId(`${name}-small`)).toHaveLength(1);
	// Wait for react morph animation to end
	await waitFor(() => {
		expect(queryAllByTestId(`${name}-full`)).toHaveLength(1);
	})
	// Close via close button
	fireEvent.click(getByTestId(`${name}-close`));
	expect(overlay).toHaveStyle('pointer-events: none');
	expect(queryAllByTestId(`${name}-full`)).toHaveLength(1);
	expect(queryAllByTestId(`${name}-small`)).toHaveLength(2);
	// Wait for react morph animation to end
	await waitFor(() => {
		expect(queryAllByTestId(`${name}-small`)).toHaveLength(1);
	})
	// Expand
	fireEvent.click(getByTestId(`${name}-expand`));
	expect(overlay).toHaveStyle('pointer-events: auto');
	expect(queryAllByTestId(`${name}-full`)).toHaveLength(2);
	expect(queryAllByTestId(`${name}-small`)).toHaveLength(1);
	await waitFor(() => {
		expect(queryAllByTestId(`${name}-full`)).toHaveLength(1);
	})
	// Close via overlay click
	fireEvent.click(overlay);
	expect(overlay).toHaveStyle('pointer-events: none');
	expect(queryAllByTestId(`${name}-full`)).toHaveLength(1);
	expect(queryAllByTestId(`${name}-small`)).toHaveLength(2);
	await waitFor(() => {
		expect(queryAllByTestId(`${name}-small`)).toHaveLength(1);
	});
};

it('matches snapshot', () => {
	const { container } = render(<FuelCost />);
	expect(container.firstChild).toMatchSnapshot();
});

it('expands and closes vehicleinfo correctly', () => {
	return testTogglingFor('vehicleinfo')
});

it('expands and closes routeselector correctly', () => {
	return testTogglingFor('routeselector')
});

it('expands and closes passengersselector correctly', () => {
	return testTogglingFor('passengersselector')
});