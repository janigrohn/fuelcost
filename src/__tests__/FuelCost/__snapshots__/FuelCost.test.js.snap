// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-fzqARJ wjjiM"
    data-testid="vehicleinfo-placeholder"
  />
  <div
    class="sc-AxjAm dekorr"
    data-testid="vehicleinfo-small"
    style="visibility: visible;"
  >
    <button
      class="sc-AxhCb bYyHJx"
      data-testid="vehicleinfo-expand"
    >
      <h2
        class="sc-AxhUy evGmHE"
        style="visibility: visible;"
      >
        <svg
          class="title-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 11v-3h1.247c.882 0 1.235.297 1.828.909.452.465 1.925 2.091 1.925 2.091h-5zm-1-3h-2.243c-.688 0-1.051.222-1.377.581-.316.348-.895.948-1.506 1.671 1.719.644 4.055.748 5.126.748v-3zm14 5.161c0-2.823-2.03-3.41-2.794-3.631-1.142-.331-1.654-.475-3.031-.794-.55-.545-2.052-2.036-2.389-2.376l-.089-.091c-.666-.679-1.421-1.269-3.172-1.269h-7.64c-.547 0-.791.456-.254.944-.534.462-1.944 1.706-2.34 2.108-1.384 1.402-2.291 2.48-2.291 4.603 0 2.461 1.361 4.258 3.179 4.332.41 1.169 1.512 2.013 2.821 2.013 1.304 0 2.403-.838 2.816-2h6.367c.413 1.162 1.512 2 2.816 2 1.308 0 2.409-.843 2.82-2.01 1.934-.056 3.181-1.505 3.181-3.829zm-18 4.039c-.662 0-1.2-.538-1.2-1.2s.538-1.2 1.2-1.2 1.2.538 1.2 1.2-.538 1.2-1.2 1.2zm12 0c-.662 0-1.2-.538-1.2-1.2s.538-1.2 1.2-1.2 1.2.538 1.2 1.2-.538 1.2-1.2 1.2zm2.832-2.15c-.399-1.188-1.509-2.05-2.832-2.05-1.327 0-2.44.868-2.836 2.062h-6.328c-.396-1.194-1.509-2.062-2.836-2.062-1.319 0-2.426.857-2.829 2.04-.586-.114-1.171-1.037-1.171-2.385 0-1.335.47-1.938 1.714-3.199.725-.735 1.31-1.209 2.263-2.026.34-.291.774-.432 1.222-.43h5.173c1.22 0 1.577.385 2.116.892.419.393 2.682 2.665 2.682 2.665s2.303.554 3.48.895c.84.243 1.35.479 1.35 1.71 0 1.196-.396 1.826-1.168 1.888z"
          />
        </svg>
        <span
          style="text-transform: capitalize;"
        >
          5.5
          L/100km 
          1.63
          â‚¬/L
        </span>
      </h2>
    </button>
  </div>
  <div
    class="sc-fzqARJ wjjiM"
    data-testid="routeselector-placeholder"
  />
  <div
    class="sc-AxmLO hBPysr"
    data-testid="routeselector-small"
    style="visibility: visible;"
  >
    <button
      class="sc-fzpans dxKTiZ"
      data-testid="routeselector-expand"
    >
      <h2
        class="sc-fzplWN deVens"
        style="visibility: visible;"
      >
        <svg
          class="title-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.002 4.999l.501 2.001 1.496-4-3.998 1.5 2.001.499zm.416-4.999c-2.53 0-4.582 2.014-4.582 4.499 0 2.524 2.004 4.248 4.582 6.501 2.578-2.253 4.582-3.978 4.582-6.501 0-2.485-2.052-4.499-4.582-4.499zm0 7.791c-1.771 0-3.207-1.436-3.207-3.207 0-1.772 1.436-3.209 3.207-3.209s3.207 1.437 3.207 3.209c0 1.771-1.436 3.207-3.207 3.207zm-.356 6.213c0-.535.434-.968.968-.968.535 0 .969.433.969.968 0 .535-.434.968-.969.968-.534 0-.968-.433-.968-.968zm4.938-4.629v10.625l-6.455 4-5.545-4-5.545 4-6.455-4v-18l6.455 4 5.545-4 1 .777v7.223h-2v-4.813l-4 2.886v3.927h-1v-3.929l-4-2.479v13.294l4 2.479v-4.365h1v4.141l4-2.885v-3.256h2v3.255l4 2.885v-4.14h1v4.365l4-2.479v-7.5c.725-.654 1.409-1.315 2-2.011zm-8.76 5.252l-1.659-.945.583-1.024 1.66.945-.584 1.024zm2.252.805c-.433 0-.855-.087-1.253-.259l.467-1.082c.25.107.514.162.786.162.222 0 .441-.037.651-.11l.388 1.112c-.334.118-.683.177-1.039.177zm-10.922-1.2c.26 0 .514-.05.754-.148l.447 1.09c-.382.157-.786.236-1.201.236-.373 0-.741-.066-1.093-.195l.407-1.105c.221.081.452.122.686.122zm5.369-2.364c.42 0 .83.082 1.217.244l-.453 1.087c-.243-.101-.5-.153-.764-.153-.23 0-.457.04-.674.119l-.401-1.108c.346-.126.708-.189 1.075-.189zm-7.97 1.168c.535 0 .969.433.969.968 0 .535-.434.968-.969.968s-.969-.433-.969-.968c0-.535.434-.968.969-.968zm4.211.56l1.639-.981.605 1.011-1.639.982-.605-1.012z"
          />
        </svg>
        <span>
          400
          km
        </span>
      </h2>
    </button>
  </div>
  <div
    class="sc-fzqARJ wjjiM"
    data-testid="passengersselector-placeholder"
  />
  <div
    class="sc-fzokOt bRPoDJ"
    data-testid="passengersselector-small"
    style="visibility: visible;"
  >
    <button
      class="sc-fzoyAV WnsRY"
      data-testid="passengersselector-expand"
    >
      <h2
        class="sc-fzoLag pTNRQ"
        style="visibility: visible;"
      >
        <svg
          class="title-icon"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.118 16.064c2.293-.529 4.428-.993 3.394-2.945-3.146-5.942-.834-9.119 2.488-9.119 3.388 0 5.644 3.299 2.488 9.119-1.065 1.964 1.149 2.427 3.394 2.945 1.986.459 2.118 1.43 2.118 3.111l-.003.825h-15.994c0-2.196-.176-3.407 2.115-3.936zm-10.116 3.936h6.001c-.028-6.542 2.995-3.697 2.995-8.901 0-2.009-1.311-3.099-2.998-3.099-2.492 0-4.226 2.383-1.866 6.839.775 1.464-.825 1.812-2.545 2.209-1.49.344-1.589 1.072-1.589 2.333l.002.619z"
          />
        </svg>
        <span>
          1
           
          person
        </span>
      </h2>
    </button>
  </div>
</div>
`;
